api/src/main/java/org/openmrs/OrderType.java

```diff
@@ -30,13 +50,24 @@
 	
 	public static final String TEST_ORDER_TYPE_UUID = "52a447d3-a64a-11e3-9aeb-50e549534c5e";
 	
+	@Id
+	@Column(name = "order_type_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY, generator = "order_type_id_gen")
+	@SequenceGenerator(name = "order_type_id_gen", sequenceName = "order_type_order_type_id_seq")
 	private Integer orderTypeId;
 	
+	@Column(name = "java_class_name", nullable = false)
 	private String javaClassName;
 	
+	@ManyToOne
+	@JoinColumn(name = "parent")
 	private OrderType parent;
 	
 	@Independent
+	@ManyToMany
+	@LazyCollection(LazyCollectionOption.TRUE)
+	@JoinTable(name = "order_type_class_map", joinColumns = @JoinColumn(name = "order_type_id", updatable = false, insertable = false), 
+		inverseJoinColumns = @JoinColumn(name = "concept_class_id", unique = true))
 	private Collection<ConceptClass> conceptClasses;
```

Comment:
@alinmihaila99, you could probably transform and specify `conceptClasses` into a `Set` and provide the getter/setter in a backwards compatible way. This means you'd maintain the old setter/getter methods but add some casting to them so implementations specifically relying on these methods in a tightly coupled way might still use them. 

Do you have any thoughts or preference on this @ibacher, @dkayiwa?